fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/ApplyRuleJob.java

```diff
@@ -82,11 +83,24 @@ public void execute() throws AnalysisException {
                 newGroupExpression.setFromRule(rule);
                 if (newPlan instanceof LogicalPlan) {
                     pushJob(new OptimizeGroupExpressionJob(newGroupExpression, context));
+                    if (!rule.getRuleType().equals(RuleType.LOGICAL_JOIN_COMMUTE)) {
+                        pushJob(new DeriveStatsJob(newGroupExpression, context));
+                    } else {
+                        // Join Commute rule doesn't change operator's expression and children
+                        // Therefore, the stats can't be changed. Don't derive stats for it
+                        groupExpression.setStatDerived(true);
+                    }
                 } else {
                     pushJob(new CostAndEnforcerJob(newGroupExpression, context));
+                    if (newGroupExpression.children().stream().anyMatch(g -> g.getLogicalExpressions().isEmpty())) {
+                        // Only the implementation rule creates a new group. The groupExpression should be derived
```

Comment:
“Only the implementation rule creates a new group”  this comments is not always right, please update it